### DISCUSSION TITLE
wezterm_cli_path problem with wezterm and WSL

### DISCUSSION THREAD

**Discussion by MrRothstein** [mrjones2014/smart-splits.nvim#248] [Q&A] [answered]
It doesn't seem like the plugin is correctly setting the wezterm_cli_path by default. In lazy vim, I can only get it to work with the following config:
```
return {
  "mrjones2014/smart-splits.nvim",
  lazy = false,
  opts = {
    wezterm_cli_path = "wezterm.exe",
  },
}
```
If I don't set wezterm_cli_path this way, I'm able switch into a vi tab, but not out of it (it just wraps).

**Comment by mrjones2014** [comment, marked as answer]
Does anything from [this discussion](https://github.com/mrjones2014/smart-splits.nvim/discussions/221) help? 

Unfortunately I do not have any Windows machines so I can't really help debug issues like this with WSL.

**Reply by MrRothstein** [reply]
I have it set in the path of both host and vm. And I'm able to actually execute "wezterm.exe" commands with lua from the minibuffer. And it works when I manually set the option. I'm guessing the logic to default it to the windows executable is off somehow, but I can't tell what it is... is_WSL check appears to be true (at least when I execute it manually from the minibuffer). Maybe the new_config picks up a value for wezterm_cli_path before it's checked here: https://github.com/mrjones2014/smart-splits.nvim/blob/22ca8bf770965306082fe23dd22822134bf3b70a/lua/smart-splits/config.lua#L119 that's as far as I got...

**Reply by MrRothstein** [reply]
I just reread that thread and I completely missed the first comment... Yes, calling setup manually works

**Reply by jayhathaway** [reply]
> I just reread that thread and I completely missed the first comment... Yes, calling setup manually works  

Are you running Wezterm nightly or stable? Trying to figure out why it won't work on wsl2 for me.

**Comment by MrRothstein** [comment]
I think I got it... I don't think lazyvim is calling the setup function at all. When I add an explicit call to it, it works as expected:

```
return {
  "mrjones2014/smart-splits.nvim",
  lazy = false,
  config = function()
    require("smart-splits").setup({})
  end,
}
```

**Reply by mrjones2014** [reply]
I believe in lazy.nvim you can also just do `config = true` and it's equivalent

**Comment by mrjones2014** [comment]
Also considering this has come up a couple times now, I think maybe I can make the experience out of the box a bit better by lazily evaluating that config option

**Comment by mrjones2014** [comment]
Can yall check if this PR helps? https://github.com/mrjones2014/smart-splits.nvim/pull/267

cc @jayhathaway @MrRothstein

**Reply by jayhathaway** [reply]
It doesn't work for me, but it may be something else in my neovim or wezterm configs. I am using the starter package for lazy vim. I can share my wezterm config file too if needed.

**Reply by jayhathaway** [reply]
@MrRothstein Since you got it to work in the end, would you mind telling me anything about your configuration?

**Reply by MrRothstein** [reply]
Just getting lazy to call setup got it working for me. Here is my lazy config:

https://github.com/MrRothstein/lazyvim/blob/main/lua/plugins/smart-splits.lua

**Reply by jayhathaway** [reply]
> Just getting lazy to call setup got it working for me. Here is my lazy config:
> 
> https://github.com/MrRothstein/lazyvim/blob/main/lua/plugins/smart-splits.lua

Thanks I'll take a look. Would you mind sharing your windows .wezterm.lua ?

**Reply by jayhathaway** [reply]
@MrRothstein @mrjones2014 MrRothstein lazy config for the plugin worked for me!
